### DISCUSSION TITLE
How can the remote access clients communicate with each other?

### DISCUSSION THREAD

**Discussion by wycgi520** [strongswan/strongswan#2395] [Q&A]
## question
I use the remote access mode to connect clients (mobile phones, computers, etc.) to a remote server deployed with strongswan, and assign virtual addresses. However, these clients cannot communicate with each other. Can you please advise me on what additional configuration is needed to enable communication between clients

### OS
NAME="Alibaba Cloud Linux"
VERSION="3 (OpenAnolis Edition)"
ID="alinux"
ID_LIKE="rhel fedora centos anolis"
VERSION_ID="3"
VARIANT="OpenAnolis Edition"
VARIANT_ID="openanolis"
ALINUX_MINOR_ID="2104"
ALINUX_UPDATE_ID="10"
PLATFORM_ID="platform:al8"
PRETTY_NAME="Alibaba Cloud Linux 3.2104 U10 (OpenAnolis Edition)"
ANSI_COLOR="0;31"
HOME_URL="https://www.aliyun.com/"

### strongswan
Linux strongSwan U5.9.10/K5.10.134-17.1.al8.x86_64
University of Applied Sciences Rapperswil, Switzerland

### swanctl.conf
```conf
connections {
    ikev2-eap-mschapv2 {
        version = 2
        unique = never
        rekey_time = 0s
        fragmentation = yes
        dpd_delay = 60s
        send_cert = always
        pools = ipv4-addrs
        proposals = aes256-sha256-prfsha256-modp2048, aes256gcm16-prfsha384-modp1024, default
        local_addrs = %any
        local {
            certs = cert.pem
            id = www.wycgi.cn
        }
        remote {
            auth = eap-mschapv2
            eap_id = %any
        }
        children {
            ikev2-eap-mschapv2 {
                local_ts = 10.10.0.0/24
                remote_ts = 10.10.0.0/24
                rekey_time = 0s
                dpd_action = clear
                esp_proposals = aes256-sha256, aes128-sha1, default
            }
        }
    }
}
pools {
    ipv4-addrs {
        addrs = 10.10.0.0/24
        dns = 8.8.8.8,1.1.1.1
    }
}
secrets {
    private-wgy {
        file = privkey.pem
    }
    eap-user1 {
        id = user1
        secret = "123456"
    }
    eap-user2 {
        id = user2
        secret = "123456"
    }
}

```

### firewalld
systemctl enable firewalld.service

systemctl start firewalld.service

firewall-cmd --permanent --add-service="ipsec"

firewall-cmd --permanent --add-port=500/udp

firewall-cmd --permanent --add-port=4500/udp

firewall-cmd --permanent --add-masquerade

**Comment by strongX509** [comment]
The configuration parameter **`local_ts = 10.10.0.0/24`** seems sufficient for the clients receiving dynamic IP addresses from the pool **`10.10.0.0/24`** to communicate with each other. The configuration parameter **`remote_ts = 10.10.0.0/24`** can be omitted because by default the IP address assigned from the pool is used for the remote network.

The question is how you configure your clients. With a remote network of **`10.10.0.0/24`**? A strongSwan log file showing the IKEv2 negotiation with a client would be helpful for debugging your problem.

**Comment by TomHsiung** [comment]
I remember I could successfully ping my remote clients. Maybe you can draw a packet flow draft to figure out the configuration of security policy and routing rules.
